core/grants.go

```diff
@@ -195,3 +195,54 @@ func (a *API) revokeGrant(ctx context.Context, x apiGrant) error {
 
 	return nil
 }
+
+func (a *API) revokeGrantsByAccessToken(ctx context.Context, token string) error {
+	for _, p := range policies {
+		data, err := a.raftDB.Get(ctx, grantPrefix+p)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+
+		if data == nil {
```

Comment:
I dunno, the rest of the function gracefully does nothing when data is nil. Are you thinking about performance? Or readability? I'd be surprised if it's appreciably faster this way.